core/src/main/java/com/sequenceiq/cloudbreak/service/StackCommonService.java

```diff
@@ -475,4 +476,18 @@ void validateNetworkScaleRequest(StackDto stack, NetworkScaleV4Request stackNetw
     public List<SubnetIdWithResourceNameAndCrn> getAllUsedSubnetsByEnvironmentCrn(String environmentCrn) {
         return instanceMetaDataService.findAllUsedSubnetsByEnvironmentCrn(environmentCrn);
     }
+
+    public FlowIdentifier putDeleteVolumesInWorkspace(NameOrCrn nameOrCrn, String accountId, StackDeleteVolumesRequest deleteRequest) {
+        StackView stackView = stackDtoService.getStackViewByNameOrCrn(nameOrCrn, accountId);
+        Stack stack = stackService.getByIdWithLists(stackView.getId());
+        MDCBuilder.buildMdcContext(stackView);
+        LOGGER.debug("Validating Stack Delete Volumes Request for Stack ID {}", stackView.getId());
+        validateDeleteVolumesRequest(stack, deleteRequest);
+        return clusterCommonService.putDeleteVolumes(stackView.getResourceCrn(), deleteRequest);
+    }
+
+    private void validateDeleteVolumesRequest(Stack stack, StackDeleteVolumesRequest deleteRequest) {
+        verticalScalingValidatorService.validateProvider(stack, "Deleting volumes", false);
```

Comment:
Why are we validating that vertical scaling is supported? We should check if delete volume is supported. Even if vertical scaling is supported, delete volume might not be supported.

Reply:
done